agreementbot/secrets/vault/vault.go

```diff
@@ -19,59 +20,95 @@ func init() {
 
 // The fields in this object are initialized in the Initialize method in this package.
 type AgbotVaultSecrets struct {
-	token string   // The identity of this agbot in the vault
+	token      string       // The identity of this agbot in the vault
 	httpClient *http.Client // A cached http client to use for invoking the vault
-	cfg *config.HorizonConfig
+	cfg        *config.HorizonConfig
 }
 
 func (vs *AgbotVaultSecrets) String() string {
 	return fmt.Sprintf("Token: %v", vs.token)
 }
 
-func (vs *AgbotVaultSecrets) ListOrgSecret(user, password, org, name string) ([]string, error) {
-	return []string{}, nil
+// This utility will be available to any users within the org.
+func (vs *AgbotVaultSecrets) ListOrgSecret(user, password, org, name string) (map[string]string, error) {
+
+	glog.V(3).Infof(vaultPluginLogString(fmt.Sprintf("list secret %v for org %v", name, org)))
+
+	// Login the user to ensure that the vault ACLs can take effect
+	userVaultToken, err := vs.loginUser(user, password, org)
+	if err != nil {
+		return nil, secrets.ErrorResponse{Msg: fmt.Sprintf("Unable to login user %s, error: %v", user, err), Details: "", RespCode: http.StatusUnauthorized}
+	}
+
+	url := fmt.Sprintf("%s/v1/openhorizon/%s/%s", vs.cfg.GetAgbotVaultURL(), org, name)
+
+	resp, err := vs.invokeVaultWithRetry(userVaultToken, url, http.MethodGet, nil)
+	if err != nil {
+		return nil, secrets.ErrorResponse{Msg: fmt.Sprintf("Unable to list %s secret %s, error: %v", org, name, err), Details: "", RespCode: http.StatusServiceUnavailable}
+	}
+
+	defer resp.Body.Close()
```

Comment:
Some times the response body can be non-nil in the error case. So please do the close() after line 45 as 
if resp != nil && resp.Body != nil {
    defer resp.Body.Close()
}

Reply:
Hey @linggao , will that be required since `invokeVaultWithRetry` does not introduce a error case where the response body is non nil. Also, I checked the documentation for [httpClient.Do](https://golang.org/pkg/net/http/#Client.Do) and they mention that in the case both are non nil the response body is still closed.

Reply:
http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/index.html#close_http_resp_body